src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extension.cs

```diff
@@ -63,6 +64,321 @@ internal override string ExtensionName
+                var callingConvention = functionPointer.Signature.CallingConvention switch
+                {
+                    Cci.CallingConvention.Default => null, // managed is the default
+                    Cci.CallingConvention.Unmanaged => "unmanaged ",
+                    Cci.CallingConvention.CDecl => "unmanaged cdecl ",
+                    Cci.CallingConvention.Standard => "unmanaged stdcall ",
+                    Cci.CallingConvention.ThisCall => "unmanaged thiscall ",
+                    Cci.CallingConvention.FastCall => "unmanaged fastcall ",
+                    _ => throw ExceptionUtilities.UnexpectedValue(functionPointer.Signature.CallingConvention)
+                };
+
+                builder.Append(callingConvention);
+
+                appendType(functionPointer.Signature.ReturnType, builder);
+                if (functionPointer.Signature.RefKind != RefKind.None)
+                {
+                    builder.Append('&');
+                    appendModifiers(functionPointer.Signature.RefCustomModifiers, builder);
+                }
+                else
+                {
+                    appendModifiers(functionPointer.Signature.ReturnTypeWithAnnotations.CustomModifiers, builder);
+                }
+
+                builder.Append(" *(");
+                var parameters = functionPointer.Signature.Parameters;
+                for (int i = 0; i < parameters.Length; i++)
+                {
+                    if (i > 0)
+                    {
+                        builder.Append(", ");
+                    }
+
+                    ParameterSymbol parameter = parameters[i];
+                    appendType(parameter.Type, builder);
+                    if (parameter.RefKind != RefKind.None)
+                    {
+                        builder.Append('&');
+                        appendModifiers(parameter.RefCustomModifiers, builder);
+                    }
+                }
+
+                builder.Append(')');
+            }
+
+            static void appendModifiers(ImmutableArray<CustomModifier> customModifiers, StringBuilder builder)
+            {
+                var modifierStrings = ArrayBuilder<string>.GetInstance(customModifiers.Length);
+
+                foreach (CustomModifier modifier in customModifiers)
+                {
+                    var modifierBuilder = PooledStringBuilder.GetInstance();
+                    modifierBuilder.Builder.Append(modifier.IsOptional ? " modopt(" : " modreq(");
+
+                    appendType(((PublicModel.NamedTypeSymbol)modifier.Modifier).UnderlyingTypeSymbol, modifierBuilder.Builder);
+                    modifierBuilder.Builder.Append(')');
+
+                    modifierStrings.Add(modifierBuilder.ToStringAndFree());
+                }
+
+                modifierStrings.Sort(StringComparer.Ordinal); // Actual order doesn't matter - just want to be deterministic
```

Comment:
>modifierStrings.Sort(StringComparer.Ordinal); [](http://example.com/codeflow?start=16&length=45)

In metadata order of custom modifiers is significant. like order of parameters and we are not sorting parameters.